### PR TITLE
Typos: `SwiftCompilerSources`

### PR DESCRIPTION
**Pull request by Sajjon**
Fix typos in `SwiftCompilerSources`

This is one batch of many PRs fixing typos, [see the tracking issue.](https://github.com/swiftlang/swift/issues/75022)

### CODE REVIEW

**Review by AnthonyLatsis — COMMENTED** [review]
Just one suggestion, otherwise looks good.

File: SwiftCompilerSources/Sources/Optimizer/FunctionPasses/AsyncDemotion.swift [hunk 1]
```diff
@@ -590,7 +590,7 @@ private func analyzeForDemotion(_ function: Function) -> Node? {
   // because it may be a caller of a function that can be demoted.
 
   var executors = Set<Executor>()
-  var knownAsyncApplys: [ApplySite] = []
+  var knownAsyncApplies: [ApplySite] = []
```
Comment (AnthonyLatsis) [inline SwiftCompilerSources/Sources/Optimizer/FunctionPasses/AsyncDemotion.swift:593]: I would call this `knownAsyncApplySites` instead.

Comment (Sajjon) [reply SwiftCompilerSources/Sources/Optimizer/FunctionPasses/AsyncDemotion.swift]: Fixed!

**Review by AnthonyLatsis — APPROVED** [review]
Thanks! Please consider squashing post review commits at this stage. Although we can do that by selecting the merge strategy on GitHub when it is appropriate to squash the whole thing, code owners are not always mindful about this.

**Comment by Sajjon** [comment]
@AnthonyLatsis I've update to main branch, solve a merge conflict and squashed into a single commit. So re-review ready!

**Review by eeckstein — APPROVED** [review]

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

File: SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift [hunk 2]
```diff
@@ -112,7 +112,7 @@ private func extendAccessScopes(dependence: LifetimeDependence,
 /// caller scope, which is handled separately). A nested 'read' access can never interfere with another access in the
 /// same outer 'read', because it is impossible to nest a 'modify' access within a 'read'. For 'modify' accesses,
 /// however, the inner scope must be extended for correctness. A 'modify' access can interfere with other 'modify'
-/// accesss in the same scope. We rely on exclusivity diagnostics to report these interferences. For example:
+/// access in the same scope. We rely on exclusivity diagnostics to report these interferences. For example:
```
Comment (xwu) [inline SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift:115]: I think the intended word here would have been "accesses."

**Pull request merged**